### ISSUE TITLE
All panics in TodoMVC

### ISSUE THREAD

**Issue by Type1J** [sycamore-rs/sycamore#174]
**Describe the bug**
All panics in TodoMVC when there's no completed items, but some uncompleted items.

**To Reproduce**
Run TodoMVC example,
Add an item
Click "All"
See in script console that a panic occured.

**Expected behavior**
TodoMVC should never panic

**Environment**

- Sycamore: master
- Browser: Chrome
- OS: Windows

**Comment by lukechu10** [comment]
I can't seem to reproduce. Are you running in debug mode? What is exactly the panic message that you see?

**Comment by Type1J** [comment]
I'm using `trunk serve` to build and run. Here's what the console output is:

```
﻿
index-d2a7be0e8db3223d.js:275 panicked at 'start is the first index where items[start] != new_items[start]', C:\Projects\test_sycamore\sycamore\packages\sycamore\src\rx\iter.rs:69:17

Stack:

Error
    at imports.wbg.__wbg_new_59cb74e423758ede (http://127.0.0.1:8080/index-d2a7be0e8db3223d.js:278:15)
    at console_error_panic_hook::Error::new::h31e3bcd30a481ad9 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[5078]:0x105329)
    at console_error_panic_hook::hook_impl::h2d0324b762173116 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[534]:0x6cf37)
    at console_error_panic_hook::hook::h0543ebbcf2e671f0 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[5757]:0x10e67b)
    at core::ops::function::Fn::call::h4289eecc25507e2b (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[4566]:0xfd4ff)
    at std::panicking::rust_panic_with_hook::hbdbceb5cd158bf19 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[1275]:0x9fde6)
    at std::panicking::begin_panic::{{closure}}::hd14a16601d3f034a (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[2015]:0xc0205)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h43e62669091d7c86 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[2734]:0xd6ef0)
    at std::panicking::begin_panic::h0e18819261d1b2a7 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[3913]:0xf1ba4)
    at sycamore::rx::iter::map_keyed::{{closure}}::{{closure}}::h83d90ecb1d91c562 (http://127.0.0.1:8080/index-d2a7be0e8db3223d_bg.wasm:wasm-function[80]:0x6bca)

imports.wbg.__wbg_error_4bb6c2a97407129a	@	index-d2a7be0e8db3223d.js:275
console_error_panic_hook::error::h544127f39e431f6a	@	index-d2a7be0e8db3223d_bg.wasm:0xb3520
console_error_panic_hook::hook_impl::h2d0324b762173116	@	index-d2a7be0e8db3223d_bg.wasm:0x6d02f
console_error_panic_hook::hook::h0543ebbcf2e671f0	@	index-d2a7be0e8db3223d_bg.wasm:0x10e67b
core::ops::function::Fn::call::h4289eecc25507e2b	@	index-d2a7be0e8db3223d_bg.wasm:0xfd4ff
std::panicking::rust_panic_with_hook::hbdbceb5cd158bf19	@	index-d2a7be0e8db3223d_bg.wasm:0x9fde6
std::panicking::begin_panic::{{closure}}::hd14a16601d3f034a	@	index-d2a7be0e8db3223d_bg.wasm:0xc0205
std::sys_common::backtrace::__rust_end_short_backtrace::h43e62669091d7c86	@	index-d2a7be0e8db3223d_bg.wasm:0xd6ef0
std::panicking::begin_panic::h0e18819261d1b2a7	@	index-d2a7be0e8db3223d_bg.wasm:0xf1ba4
sycamore::rx::iter::map_keyed::{{closure}}::{{closure}}::h83d90ecb1d91c562	@	index-d2a7be0e8db3223d_bg.wasm:0x6bca
sycamore::rx::effect::untrack::{{closure}}::h1e55161eef6fccce	@	index-d2a7be0e8db3223d_bg.wasm:0x69f9d
std::thread::local::LocalKey<T>::try_with::haf71332820944c2a	@	index-d2a7be0e8db3223d_bg.wasm:0x8fac9
sycamore::rx::effect::untrack::h2702b85e677f3133	@	index-d2a7be0e8db3223d_bg.wasm:0x3942b
sycamore::rx::iter::map_keyed::{{closure}}::h3f52e5df53175604	@	index-d2a7be0e8db3223d_bg.wasm:0x9e470
sycamore::flow::Keyed<G>::__create_component::{{closure}}::hf6eabd0d3796a160	@	index-d2a7be0e8db3223d_bg.wasm:0xfa411
sycamore::rx::effect::create_selector_with::{{closure}}::{{closure}}::hb4b250f22b0008cf	@	index-d2a7be0e8db3223d_bg.wasm:0x50145
<alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut::ha866ef6f8a59b9fb	@	index-d2a7be0e8db3223d_bg.wasm:0xf700c
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::{{closure}}::h5a7e55ca90eb1f62	@	index-d2a7be0e8db3223d_bg.wasm:0xf5651
core::ops::function::FnOnce::call_once{{vtable.shim}}::hfd58993b5e8657d5	@	index-d2a7be0e8db3223d_bg.wasm:0x105df2
<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h1b4450b9c0fef9b5	@	index-d2a7be0e8db3223d_bg.wasm:0xd6ca5
sycamore::rx::create_root::internal::{{closure}}::h01c170fbfe0737b4	@	index-d2a7be0e8db3223d_bg.wasm:0x76ad3
std::thread::local::LocalKey<T>::try_with::h2b008f1165e3d0f3	@	index-d2a7be0e8db3223d_bg.wasm:0x6ec68
std::thread::local::LocalKey<T>::with::h4d7c499841141787	@	index-d2a7be0e8db3223d_bg.wasm:0xd53d9
sycamore::rx::create_root::internal::h9f2d182d4c650899	@	index-d2a7be0e8db3223d_bg.wasm:0xdee83
sycamore::rx::create_root::hd1d0ebc2948980be	@	index-d2a7be0e8db3223d_bg.wasm:0xe644b
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::h1d8ef736a7fe6a04	@	index-d2a7be0e8db3223d_bg.wasm:0x116bb
std::thread::local::LocalKey<T>::try_with::heee05cb9d94a6392	@	index-d2a7be0e8db3223d_bg.wasm:0x84a4b
std::thread::local::LocalKey<T>::with::h8310632ac33468b7	@	index-d2a7be0e8db3223d_bg.wasm:0xab9ba
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::he82f5efe06b66883	@	index-d2a7be0e8db3223d_bg.wasm:0xc0030
sycamore::rx::signal::Signal<T>::trigger_subscribers::h7e876817bbf2e464	@	index-d2a7be0e8db3223d_bg.wasm:0x329b1
sycamore::rx::signal::Signal<T>::set::h04069866137a1e02	@	index-d2a7be0e8db3223d_bg.wasm:0x891aa
sycamore::rx::effect::create_selector_with::{{closure}}::{{closure}}::hfcb075e28fe9c880	@	index-d2a7be0e8db3223d_bg.wasm:0x5110d
<alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut::ha866ef6f8a59b9fb	@	index-d2a7be0e8db3223d_bg.wasm:0xf700c
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::{{closure}}::h5a7e55ca90eb1f62	@	index-d2a7be0e8db3223d_bg.wasm:0xf5651
core::ops::function::FnOnce::call_once{{vtable.shim}}::hfd58993b5e8657d5	@	index-d2a7be0e8db3223d_bg.wasm:0x105df2
<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h1b4450b9c0fef9b5	@	index-d2a7be0e8db3223d_bg.wasm:0xd6ca5
sycamore::rx::create_root::internal::{{closure}}::h01c170fbfe0737b4	@	index-d2a7be0e8db3223d_bg.wasm:0x76ad3
std::thread::local::LocalKey<T>::try_with::h2b008f1165e3d0f3	@	index-d2a7be0e8db3223d_bg.wasm:0x6ec68
std::thread::local::LocalKey<T>::with::h4d7c499841141787	@	index-d2a7be0e8db3223d_bg.wasm:0xd53d9
sycamore::rx::create_root::internal::h9f2d182d4c650899	@	index-d2a7be0e8db3223d_bg.wasm:0xdee83
sycamore::rx::create_root::hd1d0ebc2948980be	@	index-d2a7be0e8db3223d_bg.wasm:0xe644b
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::h1d8ef736a7fe6a04	@	index-d2a7be0e8db3223d_bg.wasm:0x116bb
std::thread::local::LocalKey<T>::try_with::heee05cb9d94a6392	@	index-d2a7be0e8db3223d_bg.wasm:0x84a4b
std::thread::local::LocalKey<T>::with::h8310632ac33468b7	@	index-d2a7be0e8db3223d_bg.wasm:0xab9ba
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::he82f5efe06b66883	@	index-d2a7be0e8db3223d_bg.wasm:0xc0030
sycamore::rx::signal::Signal<T>::trigger_subscribers::he903a74b6bf661da	@	index-d2a7be0e8db3223d_bg.wasm:0x336e6
sycamore::rx::signal::Signal<T>::set::h64397140e774869f	@	index-d2a7be0e8db3223d_bg.wasm:0xabe4c
todomvc::footer::Footer<G>::__create_component::{{closure}}::{{closure}}::{{closure}}::h5e905d023dab7967	@	index-d2a7be0e8db3223d_bg.wasm:0xf66aa
todomvc::footer::Footer<G>::__create_component::{{closure}}::{{closure}}::{{closure}}::h9098998af00f3506	@	index-d2a7be0e8db3223d_bg.wasm:0xdd35a
<dyn core::ops::function::Fn<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h8c96069e03f721c8	@	index-d2a7be0e8db3223d_bg.wasm:0xa6ca2
__wbg_adapter_24	@	index-d2a7be0e8db3223d.js:204
real	@	index-d2a7be0e8db3223d.js:190
index-d2a7be0e8db3223d_bg.wasm:0x119a69 Uncaught RuntimeError: unreachable
    at __rust_start_panic (index-d2a7be0e8db3223d_bg.wasm:0x119a69)
    at rust_panic (index-d2a7be0e8db3223d_bg.wasm:0x1151b6)
    at std::panicking::rust_panic_with_hook::hbdbceb5cd158bf19 (index-d2a7be0e8db3223d_bg.wasm:0x9fe0a)
    at std::panicking::begin_panic::{{closure}}::hd14a16601d3f034a (index-d2a7be0e8db3223d_bg.wasm:0xc0205)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h43e62669091d7c86 (index-d2a7be0e8db3223d_bg.wasm:0xd6ef0)
    at std::panicking::begin_panic::h0e18819261d1b2a7 (index-d2a7be0e8db3223d_bg.wasm:0xf1ba4)
    at sycamore::rx::iter::map_keyed::{{closure}}::{{closure}}::h83d90ecb1d91c562 (index-d2a7be0e8db3223d_bg.wasm:0x6bca)
    at sycamore::rx::effect::untrack::{{closure}}::h1e55161eef6fccce (index-d2a7be0e8db3223d_bg.wasm:0x69f9d)
    at std::thread::local::LocalKey<T>::try_with::haf71332820944c2a (index-d2a7be0e8db3223d_bg.wasm:0x8fac9)
    at sycamore::rx::effect::untrack::h2702b85e677f3133 (index-d2a7be0e8db3223d_bg.wasm:0x3942b)
__rust_start_panic	@	index-d2a7be0e8db3223d_bg.wasm:0x119a69
rust_panic	@	index-d2a7be0e8db3223d_bg.wasm:0x1151b6
std::panicking::rust_panic_with_hook::hbdbceb5cd158bf19	@	index-d2a7be0e8db3223d_bg.wasm:0x9fe0a
std::panicking::begin_panic::{{closure}}::hd14a16601d3f034a	@	index-d2a7be0e8db3223d_bg.wasm:0xc0205
std::sys_common::backtrace::__rust_end_short_backtrace::h43e62669091d7c86	@	index-d2a7be0e8db3223d_bg.wasm:0xd6ef0
std::panicking::begin_panic::h0e18819261d1b2a7	@	index-d2a7be0e8db3223d_bg.wasm:0xf1ba4
sycamore::rx::iter::map_keyed::{{closure}}::{{closure}}::h83d90ecb1d91c562	@	index-d2a7be0e8db3223d_bg.wasm:0x6bca
sycamore::rx::effect::untrack::{{closure}}::h1e55161eef6fccce	@	index-d2a7be0e8db3223d_bg.wasm:0x69f9d
std::thread::local::LocalKey<T>::try_with::haf71332820944c2a	@	index-d2a7be0e8db3223d_bg.wasm:0x8fac9
sycamore::rx::effect::untrack::h2702b85e677f3133	@	index-d2a7be0e8db3223d_bg.wasm:0x3942b
sycamore::rx::iter::map_keyed::{{closure}}::h3f52e5df53175604	@	index-d2a7be0e8db3223d_bg.wasm:0x9e470
sycamore::flow::Keyed<G>::__create_component::{{closure}}::hf6eabd0d3796a160	@	index-d2a7be0e8db3223d_bg.wasm:0xfa411
sycamore::rx::effect::create_selector_with::{{closure}}::{{closure}}::hb4b250f22b0008cf	@	index-d2a7be0e8db3223d_bg.wasm:0x50145
<alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut::ha866ef6f8a59b9fb	@	index-d2a7be0e8db3223d_bg.wasm:0xf700c
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::{{closure}}::h5a7e55ca90eb1f62	@	index-d2a7be0e8db3223d_bg.wasm:0xf5651
core::ops::function::FnOnce::call_once{{vtable.shim}}::hfd58993b5e8657d5	@	index-d2a7be0e8db3223d_bg.wasm:0x105df2
<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h1b4450b9c0fef9b5	@	index-d2a7be0e8db3223d_bg.wasm:0xd6ca5
sycamore::rx::create_root::internal::{{closure}}::h01c170fbfe0737b4	@	index-d2a7be0e8db3223d_bg.wasm:0x76ad3
std::thread::local::LocalKey<T>::try_with::h2b008f1165e3d0f3	@	index-d2a7be0e8db3223d_bg.wasm:0x6ec68
std::thread::local::LocalKey<T>::with::h4d7c499841141787	@	index-d2a7be0e8db3223d_bg.wasm:0xd53d9
sycamore::rx::create_root::internal::h9f2d182d4c650899	@	index-d2a7be0e8db3223d_bg.wasm:0xdee83
sycamore::rx::create_root::hd1d0ebc2948980be	@	index-d2a7be0e8db3223d_bg.wasm:0xe644b
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::h1d8ef736a7fe6a04	@	index-d2a7be0e8db3223d_bg.wasm:0x116bb
std::thread::local::LocalKey<T>::try_with::heee05cb9d94a6392	@	index-d2a7be0e8db3223d_bg.wasm:0x84a4b
std::thread::local::LocalKey<T>::with::h8310632ac33468b7	@	index-d2a7be0e8db3223d_bg.wasm:0xab9ba
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::he82f5efe06b66883	@	index-d2a7be0e8db3223d_bg.wasm:0xc0030
sycamore::rx::signal::Signal<T>::trigger_subscribers::h7e876817bbf2e464	@	index-d2a7be0e8db3223d_bg.wasm:0x329b1
sycamore::rx::signal::Signal<T>::set::h04069866137a1e02	@	index-d2a7be0e8db3223d_bg.wasm:0x891aa
sycamore::rx::effect::create_selector_with::{{closure}}::{{closure}}::hfcb075e28fe9c880	@	index-d2a7be0e8db3223d_bg.wasm:0x5110d
<alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut::ha866ef6f8a59b9fb	@	index-d2a7be0e8db3223d_bg.wasm:0xf700c
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::{{closure}}::h5a7e55ca90eb1f62	@	index-d2a7be0e8db3223d_bg.wasm:0xf5651
core::ops::function::FnOnce::call_once{{vtable.shim}}::hfd58993b5e8657d5	@	index-d2a7be0e8db3223d_bg.wasm:0x105df2
<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h1b4450b9c0fef9b5	@	index-d2a7be0e8db3223d_bg.wasm:0xd6ca5
sycamore::rx::create_root::internal::{{closure}}::h01c170fbfe0737b4	@	index-d2a7be0e8db3223d_bg.wasm:0x76ad3
std::thread::local::LocalKey<T>::try_with::h2b008f1165e3d0f3	@	index-d2a7be0e8db3223d_bg.wasm:0x6ec68
std::thread::local::LocalKey<T>::with::h4d7c499841141787	@	index-d2a7be0e8db3223d_bg.wasm:0xd53d9
sycamore::rx::create_root::internal::h9f2d182d4c650899	@	index-d2a7be0e8db3223d_bg.wasm:0xdee83
sycamore::rx::create_root::hd1d0ebc2948980be	@	index-d2a7be0e8db3223d_bg.wasm:0xe644b
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::{{closure}}::h1d8ef736a7fe6a04	@	index-d2a7be0e8db3223d_bg.wasm:0x116bb
std::thread::local::LocalKey<T>::try_with::heee05cb9d94a6392	@	index-d2a7be0e8db3223d_bg.wasm:0x84a4b
std::thread::local::LocalKey<T>::with::h8310632ac33468b7	@	index-d2a7be0e8db3223d_bg.wasm:0xab9ba
sycamore::rx::effect::create_effect_initial::internal::{{closure}}::he82f5efe06b66883	@	index-d2a7be0e8db3223d_bg.wasm:0xc0030
sycamore::rx::signal::Signal<T>::trigger_subscribers::he903a74b6bf661da	@	index-d2a7be0e8db3223d_bg.wasm:0x336e6
sycamore::rx::signal::Signal<T>::set::h64397140e774869f	@	index-d2a7be0e8db3223d_bg.wasm:0xabe4c
todomvc::footer::Footer<G>::__create_component::{{closure}}::{{closure}}::{{closure}}::h5e905d023dab7967	@	index-d2a7be0e8db3223d_bg.wasm:0xf66aa
todomvc::footer::Footer<G>::__create_component::{{closure}}::{{closure}}::{{closure}}::h9098998af00f3506	@	index-d2a7be0e8db3223d_bg.wasm:0xdd35a
<dyn core::ops::function::Fn<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h8c96069e03f721c8	@	index-d2a7be0e8db3223d_bg.wasm:0xa6ca2
__wbg_adapter_24	@	index-d2a7be0e8db3223d.js:204
real	@	index-d2a7be0e8db3223d.js:190
```

**Comment by lukechu10** [comment]
I believe this is a duplicate of #172.
I recently fixed this in #173. Could you try updating your local repository with the latest changes and see if this still happens?

**Comment by Type1J** [comment]
That seems to have fixed it. Thanks!
